src/test/java/de/tum/in/www1/artemis/exam/ExamUserIntegrationTest.java

```diff
@@ -201,17 +195,23 @@ void testVerifyExamUserAttendance() throws Exception {
         headers.set("X-Artemis-Client-Fingerprint", "bar");
         headers.set("X-Forwarded-For", "10.0.28.1");
 
-        for (var studentExam : studentExams) {
+        studentExams.parallelStream().forEach(studentExam -> {
             var user = studentExam.getUser();
             database.changeUser(user.getLogin());
-            var response = request.get("/api/courses/" + course2.getId() + "/exams/" + exam2.getId() + "/student-exams/" + studentExam.getId() + "/conduction", HttpStatus.OK,
-                    StudentExam.class, headers);
+            StudentExam response = null;
+            try {
+                response = request.get("/api/courses/" + course2.getId() + "/exams/" + exam2.getId() + "/student-exams/" + studentExam.getId() + "/conduction", HttpStatus.OK,
+                        StudentExam.class, headers);
+            }
+            catch (Exception e) {
```

Comment:
Parallel streams automatically distribute the workload across multiple threads, which can help ensure that each thread is operating at maximum capacity and minimize the risk of thread starvation:)
Also its faster :D

Reply:
`studentExams` has 4 elements, a parallel stream should not make a big difference for such a small list when compared to a for-loop. I think a for-loop is the better choice here so that we can avoid a try catch alltogether.